### ISSUE TITLE
AspectRatio problem 

### ISSUE THREAD

**Issue by Angels-group** [google/ExoPlayer#1069]
Try to use ExoPlayer for OnlineTV player. 
In Huawei P6 and in Huawei Mate (vivante GPU) some problems.
In method 

@Override
    public void onVideoSizeChanged(int width, int height, int unappliedRotationDegrees, float pixelWidthAspectRatio) {
        shutterView.setVisibility(View.GONE);
        videoFrame.setAspectRatio(height == 0 ? 1 : (width * pixelWidthAspectRatio) / height);       

    }

if i use this method - some videos has dropped frames.
If i set

 videoFrame.setAspectRatio = 0

video work fine, without dropped frames, but video distorted. 
The problem manifests itself when you resize videoFrame

**Comment by ojw28** [comment]
Are these issues specific to the devices you mention? If you, you'll need to report them to Huawei.

**Comment by Angels-group** [comment]
Can't test in any another device. Android virtual device don't show any video. 
the problem appears not all streaming videos. only those who have an aspect ratio of 4:3 (as i understand)

**Comment by ojw28** [comment]
If you can provide sample content that reproduces the issue, we can try on a different device.

**Comment by Angels-group** [comment]
I have only link to the video stream, but link contains token, so i can't post link here, sorry.

can i use any another method to resize video with aspect ratio ? 
For example not use com.google.android.exoplayer.AspectRatioFrameLayout
and use another method to resize

**Comment by ojw28** [comment]
Given you don't know what the root cause of the problem is, it doesn't seem right to be trying to fix it by replacing pieces of code with other pieces of code. You need to diagnose what's actually wrong so that you make sure you're fixing the right thing, I think?

You can download the stream locally and email it as an attachment to `dev.exoplayer@gmail.com` if you'd like us to take a look. Otherwise, we don't have enough information to debug the issue.

**Comment by Angels-group** [comment]
Okay, i understand you. I'll try to download stream video, or find any another stream link with this bug. 
As soon as i find - i'll write to this e-mail. Thank you very much !

**Comment by Angels-group** [comment]
Unfortunately, the sample is still not ready.
I could give a link to stream, but it is available only on the territory of the Republic of Belarus. Restrictions on the external server traffic.

I found that the problem appears only after I change the size of the surface while playing a video. For example, when I change the screen orientation. Surface becomes larger - the problems begin. Dropped frames.
If I set setAspectRatio = 0 (in method onVideoSizeChanged) - no problems. However, the picture will be stretched without observing proportions, but work fine.

If I start viewing the stream immediately in full screen mode (without changing the size surface) - no problems.
